cms/djangoapps/api/docs/decisions/001-course-apps-api.rst

```diff
@@ -0,0 +1,153 @@
+Course Apps API
+_______________
+
+Status
+======
+Proposal
+
+Context
+=======
+
+The new `Course Authoring MFE`_ includes a new UX called "Pages and Resources"
+for configuring course "apps" such as progress, wiki, teams, discussions etc.
+all from one place. Currently these apps are intermingled with the core
+platform and are inseparable from it, however future apps could be implemented
+as plugins.
+
+In order to drive this new UI, we need a way to discover installed and available
+course apps and provide an API that can be used by MFEs to display information
+related to these apps and potentially enable/disable these apps.
+
+.. _Course Authoring MFE: https://github.com/edx/frontend-app-course-authoring/
+
+
+Decision
+========
+
+We can develop a new course apps API that can be used to manage the apps
+available for a course. The list of *available* apps might not match the list
+of *installed* apps since it is possible that certain apps may be disabled, or
+otherwise unavailable for a particular course.
+
+The course apps API will only list the available apps, and allow toggling them
+on or off. It will not support any other verbs or settings. Each app can
+provide its own API(s) for configuring app-specific settings, which will not be
+in scope here. Currently the mechanism for extending the frontend with custom
+config pages for plug-in apps is TBD.
+
+Each app will have the following associated metadata:
+
+- **id**: A unique identifier for the app. This can be used in the frontend to
+  look up the app's friendly title and description.
+- **enabled**: Is this app enabled for the current course.
+- **permissions**: Apps can potentially enable/disable certain operations, for
+  users based on their role. The permissions a user has can be listed here. Apps
+  may have their own permissions but at this very least this will include:
+
+    - **enable**: Can the current user enable/disable this app.
+    - **configure**: Can the current user can configure this app.
+
+  Apps may have their own custom fine-grained controls over what settings a
+  particular user can configure.
+
+  If an app doesn't have any configuration it can leave out the ``configure``
+  key and the UI will simply not show a configuration option for that app.
+- **legacy_link**: If available, this will point to the legacy link for
+  configuring the app. This can be provided as a fallback while the new UX is
+  still in development.
```

Comment:
If legacy is too strong this can just be external_config_url of similar.

Or this can just be a broader approach for configuring apps? In case an app wants to link out to another MFE for configuration.

Reply:
Same sort of question... do we have a use case for this at the moment?  I expected that we'd manage rollout of the new UI via waffle flags, and that the new UI wouldn't really need to know about the old one.

Reply:
There are apps like textbooks, or potentially third party apps that might have a separate MFE to configure them. So this coul be used to link out to them if uses a `external_config_url`.